runtime/lua/vim/_extui/messages.lua

```diff
@@ -101,7 +101,10 @@ local function set_virttext(type)
       local offset = tar ~= 'box' and 0
         or api.nvim_win_get_position(win)[2] + (api.nvim_win_get_config(win).border and 1 or 0)
 
-      M.box.width = math.min(o.columns, scol - offset + width)
+      if vim.api.nvim_buf_line_count(ext.bufs.box) == 1 then
+        M.box.width = math.min(o.columns, scol - offset + width)
+      end
+
```

Comment:
Thanks for looking into it, however I don't think this condition makes much sense. I think this will suffice:
```suggestion
      -- Check if adding the virt_text on this line will exceed the current 'box' width.
      M.box.width = math.max(M.box.width, math.min(o.columns, scol - offset + width))
```